pkg/webhook/servicecatalog/servicebinding/validation/handler.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"context"
+	"github.com/hashicorp/go-multierror"
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/webhookutil"
+	admissionTypes "k8s.io/api/admission/v1beta1"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// Validator is used to implement new validation logic
+type Validator interface {
+	Validate(context.Context, admission.Request, *sc.ServiceBinding, *webhookutil.TracedLogger) error
+}
+
+// AdmissionHandler handles ServiceBinding validation
+type AdmissionHandler struct {
+	decoder *admission.Decoder
+	client  client.Client
+
+	CreateValidators []Validator
+	UpdateValidators []Validator
+}
+
+var _ admission.Handler = &AdmissionHandler{}
+var _ admission.DecoderInjector = &AdmissionHandler{}
+var _ inject.Client = &AdmissionHandler{}
+
+// NewAdmissionHandler creates new AdmissionHandler and initializes validators list
+func NewAdmissionHandler() *AdmissionHandler {
+	return &AdmissionHandler{
+		CreateValidators: []Validator{&ReferenceDeletion{}},
+		UpdateValidators: []Validator{&ReferenceDeletion{}},
```

Comment:
it shouldn't be registered for update action. Remember that we are porting functionality and the behavior need to be the same as it was previously. Right now we do not want to add any new functionality or do any bug fixing.

Reply:
[Comment](https://github.com/kubernetes-incubator/service-catalog/blob/v0.1.41/plugin/pkg/admission/servicebindings/lifecycle/admission.go#L47) suggests it should be also check for update action